messagingpush/src/main/java/io/customer/messagingpush/CustomerIOFirebaseMessagingService.kt

```diff
@@ -86,8 +86,17 @@ open class CustomerIOFirebaseMessagingService : FirebaseMessagingService() {
             handleNewToken(context = context, token = token)
         }
 
+        /**
+         * Gets initialized instance of SDK. If the SDK is not initialized, it
+         * attempts to initialize with messaging push module so the module classes
+         * are initialized as well.
+         */
+        private fun getSDKInstanceOrNull(context: Context): CustomerIO? {
```

Comment:
Why was this function added?

Reply:
Whenever we initialize SDK instance using `context` from `CustomerIOFirebaseMessagingService`, we should ideally initialize push-messaging module as well. For all instances in service requesting SDK initialization, they can call this method instead of copying the code again.

Reply:
From looking at the pull request diff, it looks like the SDK is currently calling `CustomerIO.instanceOrNull(context)`. For example, in `handleNewToken` in this file, the code today is `CustomerIO.instanceOrNull(context)` but this PR is changing it to using this function. 

What problem does `CustomerIO.instanceOrNull(context, listOf(ModuleMessagingPushFCM()))` solve instead of using what the code has today: `CustomerIO.instanceOrNull(context)`? I am assuming that the SDK works as intended today, so why change this code?

Reply:
By initializing the module early, we can register activity lifecycle callback required by messaging push module whenever we initialize SDK using context from the service. So this is actually what fixes the problem for notifications received when app was in terminated state. Also, since this file is part of messaging push module, so requesting initialization of module with SDK from these files is actually the correct behavior in my opinion.

Reply:
I think that explanation you gave is really good! Can you add this explanation as a comment to the `getSDKInstanceOrNull` function explaining what problem that function solves in it's implementation?

Reply:
Sure, updated docs on `getSDKInstanceOrNull`.

Reply:
Great, thanks. 

I think we should still ship this PR if it works, but I did have an idea for a refactor. 

The purpose of `getSDKInstanceOrNull` is to make sure that Activity lifecycle callbacks get registered with the SDK on initialization. The complexity exists because Activity lifecycle callbacks are in the messaging push module. 

If we were to make the tracking module register empty Activity lifecycle callbacks during SDK initialization and then using some sort of communication mechanism (event bus, callback functions) we can make the messaging push module get code executed when the methods in the tracking module's Activity lifecycle callback get called. 

If we were to make the SDK always register Activity lifecycle callbacks during SDK initialization, then that makes me think that this could replace something such as `getSDKInstanceOrNull` and make the SDK more stable?

Reply:
Sure Levi. Since this apparently has worked well for us and fixes the bug, we'll ship this as soon as testing is complete. If you have ideas for refactoring, please feel free to create a ticket and suggest the changes in a separate PR.